source/server/admin/active_stats.js

```diff
@@ -0,0 +1,192 @@
+  document.body.appendChild(statusDiv);
+  document.body.appendChild(activeStatsPreElement);
+  loadStats();
+}
+
+/**
+ * Initiates an ajax request for the stats JSON based on the stats parameters.
+ */
+async function loadStats() {
+  const makeQueryParam = (name) => name + '=' + encodeURIComponent(
+      document.getElementById(paramIdPrefix + name).value);
+  const params = ['filter', 'type', 'histogram_buckets'];
+  const url = '/stats?format=json&usedonly&' + params.map(makeQueryParam).join('&');
+  try {
+    const response = await fetch(url);
+    const data = await response.json();
+    renderStats(data);
+  } catch (e) {
+    statusDiv.textContent = 'Error fetching ' + url + ': ' + e;
+  }
+
+  // Update stats every 5 seconds by default.
+  window.setTimeout(loadStats, 1000*loadSettingOrUseDefault('active-update-interval', 5));
+}
+
+/**
+ * Function used to sort stat records. The highest priority is update frequency,
+ * then value (higher values are likely more interesting), and finally alphabetic,
+ * in forward order.
+ *
+ * @param {!Object} a
+ * @param {!Object} b
+ * @return {number}
+ */
+function compareStatRecords(a, b) {
+  // Sort higher change-counts first.
+  if (a.change_count != b.change_count) {
+    return b.change_count - a.change_count;
+  }
+
+  // Secondarily put higher values first -- they are often more interesting.
+  if (b.value != a.value) {
+    return b.value - a.value;
+  }
+
+  // Fall back to forward alphabetic sort.
+  if (a.name < b.name) {
+    return -1;
+  }
+  if (a.name > b.name) {
+    return 1;
+  }
+  return 0;
+}
+
+/**
+ * The active display has additional settings for tweaking it -- this helper extracts numeric
+ * values from text widgets
+ *
+ * @param {string} id
+ * @param {number} defaultValue
+ * @return {number}
+ */
+function loadSettingOrUseDefault(id, defaultValue) {
+  const elt = document.getElementById(id);
+  const value = parseInt(elt.value);
+  if (Number.isNaN(value) || value <= 0) {
+    console.log('Invalid ' + id + ': invalid positive number');
+    return defaultValue;
+  }
+  return value;
+}
+
+/**
+ * Rendering function which interprets the Json response from the server, updates
+ * the most-frequently-used map, reverse-sorts by use-count, and serializes the
+ * top ordered stats into the PRE element created in initHook.
+ *
+ * @param {!Object} data
+ */
+function renderStats(data) {
+  sortedStats = [];
+  let prevStats = currentStats;
+  currentStats = new Map();
+  for (stat of data.stats) {
+    if (!stat.name) {
+      continue; // Skip histograms for now.
+    }
+    let statRecord = prevStats.get(stat.name);
+    if (statRecord) {
+      if (statRecord.value != stat.value) {
+        statRecord.value = stat.value;
+        ++statRecord.change_count;
+      }
+    } else {
+      statRecord = {name: stat.name, value: stat.value, change_count: 0};
+    }
+    currentStats.set(stat.name, statRecord);
+    sortedStats.push(statRecord);
+  }
+  prevStats = null;
```

Comment:
I don't think setting this to null does anything meaningful. It'll go out of scope, and the garbage collector will eventually deal with it. Explicit nulling won't make the gc get to it any quicker.

Reply:
you're probably right -- v8 would likely figure it was unreferenced after that anyway.

WDYT though -- maybe I should  not rebuild the map at all, but should remember stat change-counts for the entire life of the page?

E.g. if I mistype a filter it will erase the use-count map, which might be annoying.

Reply:
I did wonder about the lifespan of change counts, but I think it probably doesn't matter too much if the data gets expired on refiltering - it'll give you a more *immediate* higher-usage indicator, which sometimes would be better, sometimes would be worse, so it's a shrug for me.

Reply:
I went ahead and made the above change as I think it is better to retain the counts of previously displayed items that have now gone out-of-scope based on the filter. If the user hits browser 'refresh' it will reset that map and we'll start counting from zero which seems reasonably intuitive for now.